pkg/array/array.go

```diff
@@ -254,30 +254,55 @@ func GetPowerStoreArrays(fs fs.Interface, filePath string) (map[string]*PowerSto
 	return arrayMap, mapper, defaultArray, nil
 }
 
-// ParseVolumeID parses volume id in from CO (Kubernetes) and tries to understand what in it are PowerStore volume id, and what is ip, protocol.
-// "/" is used as a delimiter.
+// ParseVolumeID parses a volume id from the CO (Kubernetes) and tries to extract the PowerStore volume id, Global ID, and protocol.
 //
 // Example:
 //
-//	ParseVolumeID("1cd254s/192.168.0.1/scsi") will return
-//		id = "1cd254s"
-//		ip = "192.168.0.1"
+//	ParseVolumeID("1cd254s/192.168.0.1/scsi") assuming 192.168.0.1 is the IP array PSabc0123def will return
+//		localVolumeID = "1cd254s"
+//		arrayID = "PSabc0123def"
 //		protocol = "scsi"
+//		e = nil
+//
+// Example:
+//
+//	ParseVolumeID("9f840c56-96e6-4de9-b5a3-27e7c20eaa77/PSabcdef0123/scsi:9f840c56-96e6-4de9-b5a3-27e7c20eaa77/PS0123abcdef") returns
+//		localVolumeID = "9f840c56-96e6-4de9-b5a3-27e7c20eaa77"
+//		arrayID = "PSabcdef0123"
+//		protocol = "scsi"
+//		remoteVolumeID = "9f840c56-96e6-4de9-b5a3-27e7c20eaa77"
+//		remoteArrayID = "PS0123abcdef"
+//		e = nil
 //
 // This function is backwards compatible and will try to understand volume protocol even if there is no such information in volume id.
 // It will do that by querying default powerstore array passed as one of the arguments
-func ParseVolumeID(ctx context.Context, volumeID string, defaultArray *PowerStoreArray /*optional*/, cap *csi.VolumeCapability) (id string, arrayID string, protocol string, e error) {
-	if volumeID == "" {
-		return "", "", "", status.Errorf(codes.FailedPrecondition,
-			"incorrect volume id ")
+func ParseVolumeID(ctx context.Context, volumeHandle string,
```

Comment:
@lukeatdell ParseVolumeID now returns 6 values. Should we consider encapsulating those values in a struct ?

Reply:
I definitely considered that. 
I don't see any reason not to. I'll try and get that changed.

Reply:
I'm losing some dev time here these next few days due to mandatory training sessions.
I spoke with the team about this change. It'll take a bit of effort to refactor the places where this is implemented, but we will try and get it in as part of one of the future stories for the metro feature.

Reply:
Thanks. Including this in a future story sounds like a good plan.